tilequeue/command.py

```diff
@@ -2116,73 +2149,95 @@ def tilequeue_meta_tile(cfg, args):
+        return
+
+    for fetch, coord_datum in fetched_coord_data:
+        coord = coord_datum['coord']
+        if check_metatile_exists:
+            existing_data = store.read_tile(coord, zip_format, layer)
+            if existing_data is not None:
+                meta_tile_logger.metatile_already_exists(
+                    parent, job_coord, coord)
+                continue
+
+        processor = Processor(
+            coord, cfg.metatile_zoom, fetch, layer_data,
+            post_process_data, formats, cfg.buffer_cfg,
+            output_calc_mapping, cfg.max_zoom, cfg.tile_sizes)
 
         try:
-            fetched_coord_data = list(data_fetcher.fetch_tiles(coord_data))
+            processor.fetch()
+
         except Exception as e:
-            meta_tile_logger.pyramid_fetch_failed(e, parent, job_coord)
+            meta_tile_logger.tile_fetch_failed(
+                e, parent, job_coord, coord)
             continue
 
-        for fetch, coord_datum in fetched_coord_data:
-            coord = coord_datum['coord']
-            if check_metatile_exists:
-                existing_data = store.read_tile(coord, zip_format, layer)
-                if existing_data is not None:
-                    meta_tile_logger.metatile_already_exists(
-                        parent, job_coord, coord)
-                    continue
+        try:
+            formatted_tiles, _ = processor.process_tiles()
 
-            processor = Processor(
-                coord, cfg.metatile_zoom, fetch, layer_data,
-                post_process_data, formats, cfg.buffer_cfg,
-                output_calc_mapping, cfg.max_zoom, cfg.tile_sizes)
+        except Exception as e:
+            meta_tile_logger.tile_process_failed(
+                e, parent, job_coord, coord)
 
-            try:
-                processor.fetch()
+        try:
+            tiles = make_metatiles(cfg.metatile_size, formatted_tiles)
+            for tile in tiles:
+                store.write_tile(
+                    tile['tile'], tile['coord'], tile['format'],
+                    tile['layer'])
+        except Exception as e:
+            meta_tile_logger.metatile_storage_failed(
+                e, parent, job_coord, coord)
+            continue
 
-            except Exception as e:
-                meta_tile_logger.tile_fetch_failed(
-                    e, parent, job_coord, coord)
-                continue
+        meta_tile_logger.tile_processed(parent, job_coord, coord)
+
+    meta_tile_logger.end_pyramid(parent, job_coord)
+
+
+def get_batch_job_coord(coords, job_idx=None):
+    if job_idx is None:
+        job_idx_str = os.getenv('AWS_BATCH_JOB_ARRAY_INDEX')
+        if job_idx_str is None:
+            return None
+        try:
+            job_idx = int(job_idx_str)
+        except ValueError:
+            return None
+    if job_idx < 0:
+        return None
+    return next(islice(coords, job_idx, None), None)
```

Comment:
Why not do something like `coords[job_idx]`?

Reply:
Doesn't work with generators unfortunately :(

```python
>>> xs = (x for x in xrange(10))
>>> xs[3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'generator' object has no attribute '__getitem__'
```